### PR TITLE
[SYCL-MLIR][Polygeist] Support bswap builtin lowering

### PR DESCRIPTION
**Pull request by Naghasan**
Lower __builtin_bswap* to the llvm ByteSwap op.

### CODE REVIEW

File: polygeist/tools/cgeist/Test/CMakeLists.txt [hunk 1]
```diff
@@ -14,6 +14,7 @@ list(APPEND MLIR_CLANG_TEST_DEPS
   FileCheck count not
   llvm-config 
   llvm-dis
+  llvm-as
```
Comment (victor-eds) [inline polygeist/tools/cgeist/Test/CMakeLists.txt:17]: Why do we need this in `cgeist` tests?

File: polygeist/tools/cgeist/Test/Verification/builtin.cpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// RUN: cgeist -O0 %s -w -o - -S --function=* | FileCheck %s
+
+// bswap
```
Comment (victor-eds) [inline polygeist/tools/cgeist/Test/Verification/builtin.cpp:3]: NIT: We usually prepend comments with `// COM:` in test files lately.

File: polygeist/tools/cgeist/Test/Verification/builtin.cpp [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// RUN: cgeist -O0 %s -w -o - -S --function=* | FileCheck %s
```
Comment (sommerlukas) [inline polygeist/tools/cgeist/Test/Verification/builtin.cpp:1]: Do we need the `function` filter at all, if we allow all functions?

**Review by sommerlukas — APPROVED** [review]
Just one nit.

File: polygeist/tools/cgeist/Test/CMakeLists.txt (hunk 1 above)
Comment (Naghasan) [reply polygeist/tools/cgeist/Test/CMakeLists.txt:17]: `math_funcs.cpp` requires `llvm-as` and was failing on my machine as it is not build by default with `sycl-toolchain`

File: polygeist/tools/cgeist/Test/Verification/builtin.cpp (hunk 3 above)
Comment (Naghasan) [reply polygeist/tools/cgeist/Test/Verification/builtin.cpp:1]: not sure, reuse an existing polygiest test

File: polygeist/tools/cgeist/Test/Verification/builtin.cpp (hunk 2 above)
Comment (Naghasan) [reply polygeist/tools/cgeist/Test/Verification/builtin.cpp]: fixed

File: polygeist/tools/cgeist/Test/Verification/builtin.cpp (hunk 3 above)
Comment (victor-eds) [reply polygeist/tools/cgeist/Test/Verification/builtin.cpp:1]: Yep, it's needed unfortunately

**Pull request merged**